drivers/volume/kdmp/kdmp.go

```diff
@@ -0,0 +1,297 @@
+	volumeInfos := make([]*storkapi.ApplicationBackupVolumeInfo, 0)
+	for _, pvc := range pvcs {
+		if pvc.DeletionTimestamp != nil {
+			log.ApplicationBackupLog(backup).Warnf("Ignoring PVC %v which is being deleted", pvc.Name)
+			continue
+		}
+		volumeInfo := &storkapi.ApplicationBackupVolumeInfo{}
+		volumeInfo.PersistentVolumeClaim = pvc.Name
+		volumeInfo.Namespace = pvc.Namespace
+		volumeInfo.DriverName = driverName
+		volume, err := core.Instance().GetVolumeForPersistentVolumeClaim(&pvc)
+		if err != nil {
+			return nil, fmt.Errorf("error getting volume for PVC: %v", err)
+		}
+		volumeInfo.Volume = volume
+
+		// create kdmp cr
+		dataExport := &kdmpapi.DataExport{}
+		dataExport.Name = backup.Name + "-" + pvc.Namespace + "-" + pvc.Name
+		dataExport.Namespace = pvc.Namespace
+		// TODO: this needs to be generic, maybe read it from configmap
+		dataExport.Spec.Type = kdmpapi.DataExportKopia
+		dataExport.Spec.Destination = kdmpapi.DataExportObjectReference{
+			Kind:       reflect.TypeOf(storkapi.BackupLocation{}).Name(),
+			Name:       backup.Spec.BackupLocation,
+			Namespace:  backup.Namespace,
+			APIVersion: StorkAPIVersion,
+		}
+		dataExport.Spec.Source = kdmpapi.DataExportObjectReference{
+			Kind:       pvc.Kind,
+			Name:       pvc.Name,
+			Namespace:  pvc.Namespace,
+			APIVersion: pvc.APIVersion,
+		}
+		_, err = kdmpShedOps.Instance().CreateDataExport(dataExport)
+		if err != nil {
+			logrus.Errorf("failed to create DataExport CR: %v", err)
+			return volumeInfos, err
+		}
+		logrus.Debugf("adding volume info: %v", volumeInfo)
+		volumeInfos = append(volumeInfos, volumeInfo)
+	}
+
+	return volumeInfos, nil
+}
+
+func (k *kdmp) GetBackupStatus(backup *storkapi.ApplicationBackup) ([]*storkapi.ApplicationBackupVolumeInfo, error) {
+	volumeInfos := make([]*storkapi.ApplicationBackupVolumeInfo, 0)
+	for _, vInfo := range backup.Status.Volumes {
+		if vInfo.DriverName != driverName {
+			continue
+		}
+		crName := backup.Name + "-" + vInfo.Namespace + "-" + vInfo.PersistentVolumeClaim
+		dataExport, err := kdmpShedOps.Instance().GetDataExport(crName, backup.Namespace)
+		if err != nil {
+			logrus.Errorf("failed to get DataExport CR: %v", err)
+			return volumeInfos, err
+		}
+		if dataExport.Status.TransferID == "" {
+			vInfo.Status = storkapi.ApplicationBackupStatusInitial
+			vInfo.Reason = "Volume backup not started yet"
+		} else {
+			vInfo.Volume = strings.Split(dataExport.Status.TransferID, "/")[1]
+			volumeBackup, err := kdmpShedOps.Instance().GetVolumeBackup(vInfo.Volume, backup.Namespace)
+			if err != nil {
+				logrus.Errorf("failed to get volumebackup CR: %s, err:%v", vInfo.Volume, err)
+				return volumeInfos, err
+			}
+			vInfo.BackupID = volumeBackup.Status.SnapshotID
+			if isBackupActive(dataExport.Status) {
+				vInfo.Status = storkapi.ApplicationBackupStatusInProgress
+				vInfo.Reason = fmt.Sprintf("Volume backup in progress. BytesDone: %v",
+					volumeBackup.Status.ProgressPercentage)
```

Comment:
progress percentage is not BytesDone. Since we don't have the actual progress percentage nor the bytes that have been transferred, lets not add it to the Reason.